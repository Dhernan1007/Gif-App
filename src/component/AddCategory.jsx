import { useState } from "react"
import PropTypes from 'prop-types';



export const AddCategory = ({ onNewCategory }) => {

const [inputValue, setInputValue] = useState('')

const onInputChange = ({ target })=>{
    setInputValue( target.value )
}

const onSubmit = ( event )=>{
    event.preventDefault(); 
    if(inputValue.trim().length <= 1)return; 
    const letter = (inputValue.charAt(0).toUpperCase() + inputValue.slice(1).toLowerCase() ); 
   
    onNewCategory( letter ); 
    setInputValue(''); 

}

  return (

    <form onSubmit={ onSubmit } aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}

/*

aria-label="form" se agrega como atributo

aria-label es un atributo de accesibilidad en HTML utilizado para proporcionar una etiqueta 
descriptiva a un elemento de interfaz de usuario que normalmente no tiene un texto visible, como
 un botón o un enlace. El propósito principal de aria-label es mejorar la accesibilidad web al 
 proporcionar una descripción textual que las tecnologías de asistencia, como los lectores de pantalla, 
 pueden leer en voz alta a los usuarios con discapacidad visual.

Cuando se utiliza aria-label, se proporciona un texto descriptivo que se asocia con el elemento, 
lo que permite a los usuarios comprender la función o el propósito del elemento sin necesidad de
 que haya texto visible en pantalla.

*/
