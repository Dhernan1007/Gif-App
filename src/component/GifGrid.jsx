import { GifItem } from "./GifItem";
import useFetchGifs from "../hooks/useFetchGifs";
import { useState } from "react";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs(category);
  const [times, setTimes] = useState(false)

  setTimeout(() => {
    setTimes(true)
  }, 1500);

  return (
    <>
      {!times ? <> {isLoading && (
        <span className="loader">L &nbsp; ading</span>
      )}
      </>
        : <>
          <h3>{category}</h3>
          <div className="card-grid">
            {images.map((image) => (
              <GifItem key={image.id}
                {...image}
              />
            )
            )}
          </div>
        </>
      }
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}