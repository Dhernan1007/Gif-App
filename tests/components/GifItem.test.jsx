import { describe, test, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/component/GifItem";

describe('Pruebas sobre el componente GifItem', () => {

    //para fines de ejemplo
    const title = 'Samurai X'
    const url = 'https://samurai-x.com/guerrero.png'


    test(' Debe hacer match con el snapshot ', () => {

        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
      
        render( <GifItem title={title} url={url} />)
        // screen.debug();
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })
    

    test('Debe de mostrar el titulo en el componente', () => {
      
        render(<GifItem title={title} url={url} />)
        expect( screen.getByText( title ) ).toBeTruthy();  
    })
    
})
