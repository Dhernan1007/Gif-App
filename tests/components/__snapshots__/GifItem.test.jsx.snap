// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifItem  Debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Samurai X"
      src="https://samurai-x.com/guerrero.png"
    />
    <p>
      Samurai X
    </p>
  </div>
</div>
`;
